include/oxen/quic/address.hpp

```diff
@@ -341,7 +341,29 @@ namespace oxen::quic
 
         auto operator<=>(const RemoteAddress& other) const
         {
+#if defined(__ANDROID__) && __NDK_MAJOR__ < 27
+            auto ret = std::strong_ordering::equal;
+
+            if (_remote_pubkey.size() != other._remote_pubkey.size())
+            {
+                ret = _remote_pubkey.size() < other._remote_pubkey.size() ? std::strong_ordering::less
+                                                                          : std::strong_ordering::greater;
+            }
+            else
+            {
+                for (size_t i = 0; i < _remote_pubkey.size(); ++i)
+                {
+                    if (_remote_pubkey[i] != other._remote_pubkey[i])
+                    {
+                        ret = _remote_pubkey[i] < other._remote_pubkey[i] ? std::strong_ordering::less
+                                                                          : std::strong_ordering::greater;
+                        break;
+                    }
+                }
+            }
```

Comment:
I assume the problem here is that `<=>` isn't working on a vector?

I don't think it makes a lot of sense to have this `<=>` operator at all here, because inequality comparisons on a RemoteAddress don't make a lot of sense to me, *and* because explicitly deleting this... doesn't seem to cause any problems at all, so let's just go that way instead rather than using an Android workaround for a function we don't need or use:

```C++
auto operator<=>(const RemoteAddress& other) const = delete; // Explicit to mask Address's <=>
bool operator==(const RemoteAddress& other) const {
    return Address::operator==(other) && _remote_pubkey == other._remote_pubkey;
}
```

Reply:
Yep pretty much - no worries, I'll make the change now then

Reply:
Ah, looks like `libSession` uses `<=>` at the moment to sort vectors of `service_node` (which inherits from `RemoteAddress`) in order to get a unique set for the snode pool - but I can just add a custom comparator for that function  instead